packages/loot-core/src/server/budget/goaltemplates.ts

```diff
@@ -231,14 +230,20 @@ async function processTemplate(month, force, category_templates) {
     //if overwrite is not preferred, set cell to original value
     for (let l = 0; l < originalCategoryBalance.length; l++) {
       await setBudget({
```

Comment:
Correction:  This is where it is set to the original balance, but only if it has a template, had a budgeted value, and is not income.